DuckDuckGo/AutocompleteViewController.swift

```diff
@@ -125,24 +143,29 @@ class AutocompleteViewController: UIViewController {
     }
 
     func updateQuery(query: String) {
-        self.query = query
         selectedItem = -1
         cancelInFlightRequests()
-        debounce.schedule { [weak self] in
-            self?.requestSuggestions(query: query)
-        }
+        self.query = query
     }
     
     func willDismiss(with query: String) {
-        guard selectedItem != -1, selectedItem < suggestions.count else { return }
-        
+        guard suggestions.indices.contains(selectedItem) else { return }
         let suggestion = suggestions[selectedItem]
-        if let url = suggestion.url {
-            if query == url.absoluteString {
-                firePixel(selectedSuggestion: suggestion)
-            }
-        } else if query == suggestion.suggestion {
-            firePixel(selectedSuggestion: suggestion)
+        firePixelForSelectedSuggestion(suggestion)
+    }
+
+    private func firePixelForSelectedSuggestion(_ suggestion: Suggestion) {
+        switch suggestion {
+        case .phrase(phrase: let phrase):
+            Pixel.fire(pixel: .autocompleteClickPhrase)
+        case .website(url: let url):
+            Pixel.fire(pixel: .autocompleteClickWebsite)
+        case .bookmark(title: let title, url: let url, isFavorite: let isFavorite, allowedInTopHits: let allowedInTopHits):
+            Pixel.fire(pixel: isFavorite ? .autocompleteClickFavorite : .autocompleteClickBookmark)
+        case .historyEntry(title: let title, url: let url, allowedInTopHits: let allowedInTopHits):
+            Pixel.fire(pixel: .autocompleteClickHistory)
+        case .unknown(value: let value):
+            assertionFailure("Unknown suggestion")
```

Comment:
You don’t use all associated values here, so this can be cleaned up a tiny bit:
```suggestion
        case .phrase:
            Pixel.fire(pixel: .autocompleteClickPhrase)
        case .website:
            Pixel.fire(pixel: .autocompleteClickWebsite)
        case .bookmark(_, _, isFavorite: let isFavorite, _):
            Pixel.fire(pixel: isFavorite ? .autocompleteClickFavorite : .autocompleteClickBookmark)
        case .historyEntry:
            Pixel.fire(pixel: .autocompleteClickHistory)
        case .unknown:
            assertionFailure("Unknown suggestion")
        }
```